doc/adrs/041_use_wildcard_collections.adoc

```diff
@@ -0,0 +1,55 @@
+= ADR-041 - Use Wildcard Collections instead of List<Object>
+
+== Context
+
+Sirius-Components has API that expose `Function<Something, List<Object>>` attributes.
+That is used especially in Descriptors (`TreeDescription`, `ListDescription`, ...)
+
+Using `List<Object>` as the return type forces implementors to copy lists returned by existing services typed with `List<Something>`, or to create lists when using a service that returns a Collection or an Iterable.
+
+Here are some alternatives:
+
+* Using `Iterable<?>`
+** Clarifies that what's returned is only intended to be iterated over, but does not make it possible to easily know the number of elements, or if there's an element at all.
+* Using `Collection<?>`
+** Better than `List<Object>`, does not impose ordering of elements, and makes it easy to check isEmpty() or size().
+* Using `List<?>`
+** Better than `List<Object>`, conveys that ordering of elements matters, and makes it easy to check isEmpty() or size().
+
+== Decision
+
+
+=== Backend
+
+* sirius-components-trees
+** `TreeDescription`
+*** `elementsProvider`: Only used in `TreeRenderer#render()`, to transform into a `List<TreeItem>` to build a `Tree`
+*** `childrenProvider`: Only used in `TreeRenderer#renderTreeItem()`, to transform into a `List<TreeItem>` to build a `TreeItem`
+** `ForDescription`
+*** `iterableProvider`: Only used in `ForComponent`, to transform into a list
+** `GroupDescription`
+*** `semanticElementsProvider`: Only used in `GroupComponent`, to transform into a list
+** `ListDescription`
+*** `itemsProvider`: Only used in `ListComponent`, to transform into a list
+** `MultiSelectDescription`
+*** `optionsProvider`: Only used in `MultiSelectComponent`, to transform into a list
+** `PageDescription`
+*** `semanticElementsProvider`: Only used in `PageComponent`, to transform into a list
+** `RadioDescription`
+*** `optionsProvider`: Only used in `RadioComponent`, to transform into a list
+** `SelectDescription`
+*** `optionsProvider`: Only used in `SelectComponent`, to transform into a list
+
+=== Compatibility layer
+
+N/A
+
+=== Frontend
+
+N/A
```

Comment:
Those parts are not required in every ADR, you can delete them

Reply:
Okay